src/Microsoft.Health.Fhir.Shared.Api.UnitTests/Features/ActionResults/FhirResultTests.cs

```diff
@@ -62,5 +68,36 @@ public void GivenANotFoundStatus_WhenReturningAResult_ThenTheStatusCodeIsSetCorr
             Assert.Null(result.Result);
             Assert.Equal(HttpStatusCode.NotFound, result.StatusCode.GetValueOrDefault());
         }
+
+        [Fact]
+        public void GivenAFhirResult_WhenHeadersThatAlreadyExistsInResponseArePassed_ThenDuplicteHeadersAreRemoved()
+        {
+            var result = FhirResult.Gone();
+            var context = new ActionContext
+            {
+                HttpContext = new DefaultHttpContext(),
+            };
+
+            IActionResultExecutor<ObjectResult> executor = Substitute.For<IActionResultExecutor<ObjectResult>>();
+            executor.ExecuteAsync(Arg.Any<ActionContext>(), Arg.Any<ObjectResult>()).ReturnsForAnyArgs(Task.CompletedTask);
+
+            ServiceCollection collection = new ServiceCollection();
+            collection.AddSingleton<IActionResultExecutor<ObjectResult>>(executor);
+            RequestContextAccessor<IFhirRequestContext> contextAccessor = Substitute.For<RequestContextAccessor<IFhirRequestContext>>();
+            collection.AddSingleton<RequestContextAccessor<IFhirRequestContext>>(contextAccessor);
+
+            ServiceProvider provider = collection.BuildServiceProvider();
+            context.HttpContext.RequestServices = provider;
+
+            result.Headers.Add("testKey", "1");
+            context.HttpContext.Response.Headers.Add("testKey", "1");
```

Comment:
Could you change the value of the header and check its value in the assert to show the header in the result is the one posted?

Reply:
Done. Also added another header variable to verify that value is not being modified.